cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ValidationHandler.java

```diff
@@ -88,25 +94,35 @@ public void healthCheck(HttpServiceRequest request, HttpServiceResponder respond
   @POST
   @Path("v1/contexts/{context}/validations/stage")
   public void validateStage(HttpServiceRequest request, HttpServiceResponder responder,
-                            @PathParam("context") String namespace) throws IOException, AccessException {
+                            @PathParam("context") String namespace,
+                            @QueryParam("doNotSkipInvalidMacroForFunctions")
```

Comment:
on second thought, I think it is better to move this to `StageValidationRequest` instead of a query param.